### PR TITLE
Add Expo template featuring RTK 2.0

### LINKED ISSUE #90: Rewrite template example tests as "integration"-style tests per the Redux "Testing" docs page
Our docs page at https://redux.js.org/usage/writing-tests recommends an "integration"-style testing approach.  But, the example tests are currently `expect(counterReducer(state, action)).toEqual(expected)`.

There was a previous PR to the CRA+JS template that tried to switch the tests over to "integration"-style:

- https://github.com/reduxjs/cra-template-redux/pull/56

We should do something along those lines for the updated templates.

### LINKED ISSUE #91: Add RTK Query to the templates
Our templates currently just have a small "counter slice" set up.  There's an async thunk included, but it's pretty simple.

Today we want users to fetch data with RTK Query.

We should configure the templates to have RTKQ set up out of the box, and have a small example endpoint set up.  (It should probably still do mock data of some kind, but not immediately sure how.)

### LINKED ISSUE #94: Add React Native templates (standard and Expo)
We should add a couple of RN templates.

@rahsheen has a couple of existing templates at:

- https://github.com/rahsheen/react-native-template-redux-typescript
- https://github.com/rahsheen/expo-template-redux-typescript

Maybe we can migrate those over?

### PR DESCRIPTION
**Pull request by aryaemami59**
This PR:

- [X] Adds template for Expo.
- [X] Replaces examples of `createAsyncThunk` with `create.asyncThunk`
- [X] Adds examples of RTK Query using `createApi`, and auto-generated React hooks
- [X] Resolves #91
- [X] Adds examples of the new `combineSlices` API
- [X] Adds `setupListeners` API
- [X] Replaces inline selectors with selectors defined in `createSlice.selectors`
- [X] Resolves #90 by adding integration tests

What else can be done:

- I would like to at some point add an example for Reselect
- We can also add some unit tests for RTK Query

## Additional Notes:

In adapting this template for `create-expo`, several modifications were necessary. Unlike more complex scaffolding tools like `create-react-app` or `react-native init`, `create-expo` has a straightforward approach. It requires the template to be an NPM package, but with specific considerations:
  
  ### Modifications in `package.json`
  `create-expo` automatically alters the package.json file in certain ways:
  
  - Removes existing `description`, `tags`, and `repository` fields.
  - Changes `version` to `1.0.0`.
  - Sets `private` to `true`.
  - Changes `name` to name of the project.
  
For reference you can see what they do here: https://github.com/expo/expo/blob/2dc48d7130d14d475c8aa79be22bf83c2593a385/packages/create-expo/src/Template.ts#L132C3-L139C33

  ### Handling of `.gitignore`
  `create-expo`  has a specific way of processing `.gitignore` files:
  
  - It generates a default `.gitignore` if one doesn’t exist.
  - Existing `.gitignore` files are replaced with this default.
  - If a `gitignore` file is present, `create-expo` uses this instead.
  
  So basically `gitignore` becomes the new `.gitignore`. So we can use `.gitignore` to ignore lock files during development, which is why I added it.
  
  Reference: https://github.com/expo/expo/blob/b1924a0637b30e94a2b2fa82eca4d12702c3684a/packages/create-expo/src/createFileTransform.ts#L44C5-L48C6
  
This template can be tested locally by running:

```bash
npx create-expo my-app -t file:<relative or absolute path to template> --no-install
```
I suggest using `--no-install` to bypass the time-consuming Expo installations if you're primarily interested in the project's initial structure.

### CODE REVIEW

**Comment by markerikson** [comment]
I have no idea if this actually works :) I'll assume it does based on your work and the similarities to the other templates.

Does Expo add a default README when you use a template? If not, we should add one.

**Comment by aryaemami59** [comment]
@markerikson It doesn't, I'll add one.
